core/src/com/ericc/the/game/systems/logic/FieldOfViewSystem.java

```diff
@@ -81,44 +87,69 @@ public void update(float deltaTime) {
         }
     }
 
+    /**
+     * A decorator-like function to call with less arguments.
+     * @param pos a Position Component of an Entity
+     * @param fov a Field of View Component of the same Entity
+     */
     private void processEntity(PositionComponent pos, FieldOfViewComponent fov) {
-        updateFOV(visibleMapArea.top, visibleMapArea.bottom, visibleMapArea.left, visibleMapArea.right, pos.x, pos.y, fov);
+        updateFOV(pos.x, pos.y, fov);
     }
 
-    private void clearFOV(int top, int bottom, int left, int right, FieldOfViewComponent fov) {
-        for (int y = top; y >= bottom; --y) {
-            for (int x = left; x <= right; ++x) {
+    /**
+     * Clears the given fov of any Entity but restricts clearing
+     * only to visible map area (by player)
+     * to improve performance.
+     * @param fov a Field of View Component of any Entity
+     */
+    private void clearFOV(FieldOfViewComponent fov) {
+        for (int y = visibleMapArea.top; y >= visibleMapArea.bottom; --y) {
+            for (int x = visibleMapArea.left; x <= visibleMapArea.right; ++x) {
                 if (map.inBoundaries(x, y)) {
                     fov.visibility[x][y] = false;
                 }
             }
         }
     }
 
-    private void updateFOV(int top, int bottom,
-                          int left, int right,
-                          int posXentity, int posYentity,
-                          FieldOfViewComponent fov) {
-        clearFOV(top, bottom, left, right, fov);
+    /**
+     * Updates the given fov component.
+     * @param posXentity x position of an Entity
```

Comment:
posXentity to trochę dziwny casing (i wording też :p )proponuję zmianę na entityXPos, dla Y analogicznie